src/herder/Upgrades.cpp

```diff
@@ -1466,42 +1466,122 @@ ConfigUpgradeSetFrame::isValidForApply() const
+
+            valid = valid && cfg.contractCompute().ledgerMaxInstructions >=
+                                 cfg.contractCompute().txMaxInstructions;
+            break;
         case ConfigSettingID::CONFIG_SETTING_CONTRACT_HISTORICAL_DATA_V0:
+            valid = cfg.contractHistoricalData().feeHistorical1KB > 0;
+            break;
         case ConfigSettingID::CONFIG_SETTING_CONTRACT_LEDGER_COST_V0:
+            valid = cfg.contractLedgerCost().ledgerMaxReadLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_READ_ENTRIES &&
+                    cfg.contractLedgerCost().ledgerMaxReadBytes >
+                        MinimumSorobanNetworkConfig::MIN_READ_BYTES &&
+                    cfg.contractLedgerCost().ledgerMaxWriteLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_ENTRIES &&
+                    cfg.contractLedgerCost().ledgerMaxWriteBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
+                    cfg.contractLedgerCost().txMaxReadLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_READ_ENTRIES &&
+                    cfg.contractLedgerCost().txMaxReadBytes >
+                        MinimumSorobanNetworkConfig::MIN_READ_BYTES &&
+                    cfg.contractLedgerCost().txMaxWriteLedgerEntries >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_ENTRIES &&
+                    cfg.contractLedgerCost().txMaxWriteBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
+                    cfg.contractLedgerCost().feeReadLedgerEntry > 0 &&
+                    cfg.contractLedgerCost().feeWriteLedgerEntry > 0 &&
+                    cfg.contractLedgerCost().feeRead1KB > 0 &&
+                    cfg.contractLedgerCost().bucketListTargetSizeBytes > 0 &&
+                    cfg.contractLedgerCost().writeFee1KBBucketListLow > 0 &&
+                    cfg.contractLedgerCost().writeFee1KBBucketListHigh > 0 &&
+                    cfg.contractLedgerCost().bucketListWriteFeeGrowthFactor > 0;
+
+            valid =
+                valid && cfg.contractLedgerCost().ledgerMaxReadLedgerEntries >=
+                             cfg.contractLedgerCost().txMaxReadLedgerEntries;
+            valid = valid && cfg.contractLedgerCost().ledgerMaxReadBytes >=
+                                 cfg.contractLedgerCost().txMaxReadBytes;
+
+            valid =
+                valid && cfg.contractLedgerCost().ledgerMaxWriteLedgerEntries >=
+                             cfg.contractLedgerCost().txMaxWriteLedgerEntries;
+            valid = valid && cfg.contractLedgerCost().ledgerMaxWriteBytes >=
+                                 cfg.contractLedgerCost().txMaxWriteBytes;
+            break;
         case ConfigSettingID::CONFIG_SETTING_CONTRACT_META_DATA_V0:
+            valid = cfg.contractMetaData().txMaxExtendedMetaDataSizeBytes >
+                        MinimumSorobanNetworkConfig::MIN_WRITE_BYTES &&
```

Comment:
Hmm, this should actually be greater than the write size in the current/updated config. Also the lower bound threshold should probably be set separately to something like 2-2.5x of `MIN_WRITE_BYTES`.